shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Override
+    public CommonPager<NamespaceVO> listByPage(final NamespaceQuery namespaceQuery) {
+        return PageResultUtils.result(namespaceQuery.getPageParameter(), () -> namespaceMapper.countByQuery(namespaceQuery), () -> namespaceMapper.selectByQuery(namespaceQuery)
+                .stream()
+                .map(NamespaceTransfer.INSTANCE::mapToVo)
+                .collect(Collectors.toList()));
+    }
+
+    @Override
+    public String delete(final List<String> ids) {
+        if (ids.contains(AdminConstants.SYS_DEFAULT_NAMESPACE_ID)) {
+            return AdminConstants.SYS_DEFAULT_NAMESPACE_ID_DELETE;
+        }
+        List<NamespaceDO> namespaceDOS = namespaceMapper.selectByIds(ids);
+        if (CollectionUtils.isEmpty(namespaceDOS)) {
+            return AdminConstants.SYS_NAMESPACE_ID_NOT_EXIST;
+        }
+        namespaceMapper.deleteByIds(ids);
+        return ShenyuResultMessage.DELETE_SUCCESS;
+    }
+
+    @Override
+    public NamespaceVO findById(final String namespaceId) {
+        return NamespaceTransfer.INSTANCE.mapToVo(namespaceMapper.selectById(namespaceId));
+    }
+
+    @Override
+    public List<NamespaceVO> list(final String name) {
+        List<NamespaceDO> namespaceDOS = namespaceMapper.selectAll(name);
+        return namespaceDOS.stream().map(NamespaceTransfer.INSTANCE::mapToVo).collect(Collectors.toList());
+    }
+
+    private NamespaceVO create(final NamespaceDTO namespaceDTO) {
+        Timestamp currentTime = new Timestamp(System.currentTimeMillis());
+        String id = UUIDUtils.getInstance().generateShortUuid();
+        String namespaceId = NamespaceIDUtils.getInstance().generateNamespaceID();
+        NamespaceDO namespaceDO = NamespaceDO.builder()
+                .id(id)
+                .namespaceId(namespaceId)
+                .name(namespaceDTO.getName())
+                .description(namespaceDTO.getDescription())
+                .dateCreated(currentTime)
+                .dateUpdated(currentTime)
+                .build();
+        List<PluginData> pluginData = pluginService.listAll();
+        List<NamespacePluginRelDO> pluginNsRelList = pluginData.stream().map(s -> NamespacePluginRelDO.builder()
+                .id(UUIDUtils.getInstance().generateShortUuid())
+                .pluginId(s.getId())
+                .config(s.getConfig())
+                .sort(s.getSort())
+                .enabled(s.getEnabled())
+                .namespaceId(namespaceId)
+                .dateCreated(currentTime)
+                .dateUpdated(currentTime)
+                .build()).collect(Collectors.toList());
+        namespaceMapper.insert(namespaceDO);
+        namespacePluginRelMapper.batchSave(pluginNsRelList);
+        return NamespaceTransfer.INSTANCE.mapToVo(namespaceDO);
+    }
+
+    private NamespaceVO update(final NamespaceDTO namespaceDTO) {
+        if (Objects.isNull(namespaceDTO) || Objects.isNull(namespaceDTO.getNamespaceId())) {
```

Comment:
remove this or check this by validation

Reply:
It is necessary to verify that the namespaceId is not empty when updating, but when creating a namespace, the namespaceId is allowed to be empty.